### DISCUSSION TITLE
should peerDependencies be semver?

### DISCUSSION THREAD

**Discussion by jw-miaem** [system-ui/theme-ui#2196] [Q&A] [answered]
Reason I ask (before raising) is because I'm unable to get rid of 'doesn't provide ...' warning with yarn berry package.json install. I looked at the peerDependencies for @theme-ui/css for example and its "@emotion/react": "^11" and I have "@emotion/react": "^11.9.0" installed which I thought would satisfy but could be wrong

**Comment by hasparus** [comment]
Could you paste the full error message here?

I'd bet that `^11` includes `^11.9.0`

https://jubianchi.github.io/semver-check/#/^11/^11.9.0

**Reply by jw-miaem** [reply]
➤ YN0002: │ @theme-ui/theme-provider@npm:0.14.2 [be04e] doesn't provide @mdx-js/react (p0c3d4), requested by @theme-ui/mdx
➤ YN0002: │ theme-ui@npm:0.14.2 [6a068] doesn't provide @emotion/react (pde673), requested by @theme-ui/core
➤ YN0002: │ theme-ui@npm:0.14.2 [6a068] doesn't provide @emotion/react (pf0bf8), requested by @theme-ui/css
➤ YN0002: │ theme-ui@npm:0.14.2 [6a068] doesn't provide @emotion/react (p2c285), requested by @theme-ui/color-modes
➤ YN0002: │ theme-ui@npm:0.14.2 [6a068] doesn't provide @emotion/react (p42588), requested by @theme-ui/mdx
➤ YN0002: │ theme-ui@npm:0.14.2 [6a068] doesn't provide @emotion/react (p086c4), requested by @theme-ui/components
➤ YN0002: │ theme-ui@npm:0.14.2 [6a068] doesn't provide @emotion/react (p9c107), requested by @theme-ui/theme-provider
➤ YN0002: │ theme-ui@npm:0.14.2 [6a068] doesn't provide @mdx-js/react (p67d5f), requested by @theme-ui/mdx

❯ yarn explain peer-requirements pde673
➤ YN0000: theme-ui@npm:0.14.2 [6a068] doesn't provide @emotion/react, breaking the following requirements:

➤ YN0000: @theme-ui/core@npm:0.14.2 [be04e]        → ^11 ✘
➤ YN0000: @theme-ui/parse-props@npm:0.14.2 [26386] → ^11 ✘

➤ YN0000: Note: these requirements start with @theme-ui/core@npm:0.14.2 [be04e]

**Comment by hasparus** [comment, marked as answer]
Oh god. I think I get it. Sorry for that, we're still on Yarn 1 here, so we didn't notice.

Is this just a warning or is it blocking your build?

**Note for myself**: Peer deps are now transitive and need to be explicitly listed in all libraries. We need a CI check for Yarn 2 and Yarn 3 installs.

---

Edit: Created https://github.com/system-ui/theme-ui/issues/2198 to track this problem.

**Reply by jw-miaem** [reply]
nw - not blocking at all
